From 6956877babbe00938c4f92d1cadf4111209d4b60 Mon Sep 17 00:00:00 2001
From: Wang Wensheng <wangwensheng4@huawei.com>
Date: Mon, 17 Apr 2023 08:41:58 +0000
Subject: [PATCH 38/53] [Huawei] mm/sharepool: Delete tgid param for
 sp_group_get_locked

Offering: HULK
hulk inclusion
category: feature
bugzilla: N/A

--------------------------------

We use sp_group_get_locked to find and get a sp_group from the global
idr. There is nothing to do with a task and its local group. Just
delete it. If one do want to get the local group for a task, using
sp_group_get_from_mm instead.

Signed-off-by: Wang Wensheng <wangwensheng4@huawei.com>
---
 mm/share_pool.c | 27 ++++-----------------------
 1 file changed, 4 insertions(+), 23 deletions(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index a37e8e92eb3f..a3b687057983 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -969,29 +969,11 @@ static struct sp_group *sp_group_get_from_mm(struct mm_struct *mm, int spg_id)
 }
 
 /* user must call sp_group_put() after use */
-static struct sp_group *sp_group_get_locked(int tgid, int spg_id)
+static struct sp_group *sp_group_get_locked(int spg_id)
 {
-	struct sp_group *spg = NULL;
-	struct task_struct *tsk = NULL;
-	int ret = 0;
-
-	if (spg_id == SPG_ID_DEFAULT) {
-		ret = get_task(tgid, &tsk);
-		if (ret)
-			return NULL;
-
-		task_lock(tsk);
-		if (tsk->mm == NULL)
-			spg = NULL;
-		else if (tsk->mm->sp_group_master)
-			spg = tsk->mm->sp_group_master->local;
-		task_unlock(tsk);
-
-		put_task_struct(tsk);
-	} else {
-		spg = idr_find(&sp_group_idr, spg_id);
-	}
+	struct sp_group *spg;
 
+	spg = idr_find(&sp_group_idr, spg_id);
 	if (!spg || !atomic_inc_not_zero(&spg->use_count))
 		return NULL;
 
@@ -1181,7 +1163,7 @@ static struct sp_group *find_or_alloc_sp_group(int spg_id)
 {
 	struct sp_group *spg;
 
-	spg = sp_group_get_locked(current->tgid, spg_id);
+	spg = sp_group_get_locked(spg_id);
 
 	if (!spg) {
 		spg = sp_group_create(spg_id);
@@ -1193,7 +1175,6 @@ static struct sp_group *find_or_alloc_sp_group(int spg_id)
 			return ERR_PTR(-ENODEV);
 		}
 		up_read(&spg->rw_lock);
-		/* spg->use_count has increased due to sp_group_get() */
 	}
 
 	return spg;
-- 
2.17.1

